lib/dart.yaml

```diff
@@ -1,194 +1,209 @@
-          - height
-          - width
-    - no-equal-conditions
-    - no-equal-nested-conditions
-    - no-equal-switch-case
-    - no-equal-switch-expression-cases
-    - no-equal-then-else
-    - no-magic-number:
-        allowed: [ -1, 0, 1 ]
-    - no-object-declaration
-    - prefer-addition-subtraction-assignments
-    - prefer-any-or-every
-    - prefer-correct-callback-field-name:
-        name-pattern: ^on[A-Z]+
-    - prefer-correct-error-name:
-        allowed-names: [ error, exception, failure ]
-    - prefer-correct-identifier-length:
-        max-identifier-length: 30
-        min-identifier-length: 2
-    - prefer-correct-stream-return-type
-    - prefer-declaring-const-constructor:
-        ignore-abstract: true
-    - prefer-early-return:
-        ignore-if-case: true
-    - prefer-explicit-function-type
-    - prefer-explicit-type-arguments
-    - prefer-immediate-return
-    - prefer-match-file-name:
-        exclude:
-          - "test/**"
-          - "lib/**/*wm.dart"
-    - prefer-moving-to-variable:
-        allowed-duplicated-chains: 3
-    - prefer-named-boolean-parameters:
-        ignore-single: true
-        ignore-single-boolean: false
-    - prefer-null-aware-spread
-    - prefer-parentheses-with-if-null
-    - prefer-public-exception-classes
-    - prefer-return-await
-    - prefer-simpler-patterns-null-check
-    - prefer-switch-with-enums
-    - prefer-test-matchers
-    - prefer-typedefs-for-callbacks
-    - prefer-unique-test-names
-    - prefer-wildcard-pattern
-    # Style
-    - unnecessary-trailing-comma: false
+    rules:
+        # https://dcm.dev/docs/rules/common
+        - arguments-ordering:
+              last:
+                  - body
+                  - child
+                  - children
+        - parameters-ordering:
+              required: "first"
+              default: "last"
+              super: "last"
+
+        - avoid-accessing-collections-by-constant-index
+        - avoid-accessing-other-classes-private-members
+        - avoid-async-call-in-sync-function
+        - avoid-barrel-files
+        - avoid-bottom-type-in-patterns
+        - avoid-bottom-type-in-records
+        # https://github.com/dart-lang/sdk/issues/45667
+        - avoid-cascade-after-if-null
+        - avoid-collapsible-if
+        - avoid-collection-methods-with-unrelated-types
+        - avoid-declaring-call-method
+        - avoid-double-slash-imports
+        - avoid-duplicate-cascades
+        - avoid-duplicate-collection-elements
+        - avoid-duplicate-exports
+        - avoid-duplicate-initializers
+        - avoid-duplicate-map-keys
+        - avoid-duplicate-mixins
+        - avoid-duplicate-named-imports
+        - avoid-duplicate-patterns
+        - avoid-duplicate-switch-case-conditions
+        - avoid-duplicate-test-assertions
+        - avoid-dynamic
+        - avoid-equal-expressions
+        - avoid-explicit-pattern-field-name
+        - avoid-explicit-type-declaration
```

Comment:
I used to avoid reduntant getter like this:
```dart
class SomeWm extends WidgetModel<SomeWidget> {
   @override
   final ValueNotifier<int> counter = ValueNotifier(0);
}

abstract class ISomeWm extends IWidgetModel<SomeWidget> {
    ValueListenable<int> get counter;
}
```
I have to declare type explicitly. Will it work with this rule?

Reply:
Maybe create an issue?